### PR TITLE
[WIP] Store and retrieve secret_key_base from disk file

### PR DESCRIPTION
**Pull request by ereslibre**
Take it from where @HouzuoGuo left it.

### CODE REVIEW

File: kubernetes/manifest-templates/velum.yaml [hunk 1]
```diff
@@ -82,3 +90,6 @@ spec:
     - name: velum-source-code
       hostPath:
         path: ${project_dir}
+    - name: velum-secrets
+      hostPath:
+        path: /tmp/velum-secrets
```
Comment (jimmykarily) [inline kubernetes/manifest-templates/velum.yaml]: I think `/tmp` dir is cleanup using various mechanisms. If the host decides to cleanup its /tmp dir or the guest decides to do the same, then bye bye secrets. If that happens, all active sessions will become invalid, no? Do we want to persist the secrets across reboots and whatever? If yes then maybe `/tmp` is not the best choice (both on host and guest).

Comment (jimmykarily) [inline kubernetes/manifest-templates/velum.yaml]: I now see, we only set this to `/tmp` on dev and test. It shouldn't be a problem then.

File: lib/velum/secrets.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+module Velum
+  # If there is a rails key base stored on disk, return the stored one.
+  # Otherwise, create a new key base from cryptographic random pool and store it.
+  class Secrets
+    def self.read_or_create_secret_key_base(pathname)
+      pathname.dirname.mkpath
+      key_base = nil
+      if pathname.exist?
+        content = pathname.read
```
Comment (jimmykarily) [inline lib/velum/secrets.rb]: Maybe add a `.strip` just in case there is some whitespace in the file? Just to make sure the next line works.

**Comment by jimmykarily** [comment]
Is this done? The e2e tests are breaking but doesn't seem related to the changes in this PR. Lets get them green before merge though.

File: kubernetes/manifest-templates/velum.yaml (hunk 1 above)
Comment (MaximilianMeister) [inline kubernetes/manifest-templates/velum.yaml]: nevertheless, as it is rails app specific, can't we always use the rails tmp dir here? then we wouldn't need an environment variable nor a volume mount.

Comment (MaximilianMeister) [inline kubernetes/manifest-templates/velum.yaml]: this also needs a rebase now after merging #104

**Comment by ereslibre** [comment]
I will fix this ASAP, so we can remove the WIP tag.

File: kubernetes/manifest-templates/velum.yaml (hunk 1 above)
Comment (MaximilianMeister) [inline kubernetes/manifest-templates/velum.yaml]: as discussed on IRC, the rails app is only mounted persistently in the dev environment, not in production. so we still need to find a folder.

**Comment by ereslibre** [comment]
@MaximilianMeister is taking over this task on https://github.com/kubic-project/velum/tree/secret_key_base, so closing this one.

**Pull request closed without merging**